### DISCUSSION TITLE
How would you discard an episode based on certain criteria?

### DISCUSSION THREAD

**Discussion by Adamantish** [vcr/vcr#872] [Q&A]
We're interfacing with an external API which is giving us a lot of trouble with its eventual consistency model. For this reason there are some tests which retry for a short while if unsuccessful. Of course, VCR just keeps replaying the troublesome 404 response.

There are a number of workarounds we could look into but the most natural option would seem to be an option to inspect an episode before it's written to file and discard it if necessary? Eject cassette might have been an option but these cassettes are meant to contain the whole sequence of various requests involved in a test example so we can't lose the good stuff.

**Comment by mcfiredrill** [comment]
Have you looked at custom request matchers?
https://relishapp.com/vcr/vcr/v/2-0-1/docs/request-matching/register-and-use-a-custom-matcher

**Comment by Adamantish** [comment]
@mcfiredrill I think I've found what I needed. the `before_record` hook.
https://relishapp.com/vcr/vcr/docs/hooks/before-record-hook

Added this to the config:

```
  config.before_record(:discard_404s) do |interaction| 
    interaction.ignore! if interaction.response.status.code == 404
  end
```

This takes effect if a cassette use is tagged like this:

```
VCR.use_cassette("my_cassette_name", :tag => :discard_404s) do
  expect("whatever will be").to be "whatever will be"
end
```
